src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -100,8 +115,78 @@ class AmendmentTableImpl : public AmendmentTable
 void
 AmendmentTableImpl::addInitial ()
 {
-    // For each amendment this version supports, construct the AmendmentState object by calling
-    // addKnown. Set any vetoes or defaults. A pointer to the AmendmentState can be stashed
+    std::set<std::pair<uint256, std::string>> toAdd;
+
+    for (auto const& _ : detail::PreEnabledAmendments)
+    {
+        uint256 hash;
+        if (!hash.SetHex (_.first))
+        {
+            std::string const errorMsg =
+                (boost::format (
+                     "PreEnabledAmendments contains an invalid hash (expected "
+                     "a hex number). Value was: %1%") %
+                 _.first).str ();
+            throw std::runtime_error (errorMsg);
+        }
+
+        toAdd.emplace (hash, _.second);
+    }
+
+    {
+        // add the amendments from the config file
+        auto const& section = getConfig ().section (SECTION_AMENDMENTS);
+        int const numExpectedToks = 2;
+        for (auto const& _ : section.lines ())
+        {
+            boost::tokenizer<> tokenizer (_);
+            int numToks = 0;
+            uint256 id;
+            std::string friendlyName;
+            for (auto const& curTok : tokenizer)
```

Comment:
This logic is tortured.  :-D  The main reason is that you don't know the number of tokens before you start the loop.

What about this sort of thing?

{
    std::vector<std::string> tokens (tokenizer.begin(), tokenizer.end());
    auto size = tokens.size();
    if (size == 0 || size > 2)
    {
        // error return
    }
    else if (size == 1) {
        tokens.push_back (/* compute the human name*/)
    }

    auto const& id = tokens[0];
    auto const& human = tokens[1];